### PR TITLE
docs: update pull request template

### PR DESCRIPTION
**Pull request by dsadgat**
docs: Update the pull request template to add in QA requirements and Value reasonings

### Reviewers
Everyone please

---

## Changes
- Updated QA section to ensure that people mention QA in the description of PRs.
- Added Value section to ensure that PR reviewers understand the reasoning for the change being made.

## QA steps
- Will take effect after PR is merged

### CODE REVIEW

**Review by maniarathi — APPROVED** [review]
Added a couple comments but otherwise LGTM and doesn't need another review from me so approving in advance.

File: PULL_REQUEST_TEMPLATE.md
Comment (Bento007) [inline PULL_REQUEST_TEMPLATE.md:16]: Will this eliminate the need to create an issue for every PR?

Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md:16]: no, you should have an issue, but it might be that you have 10 PRs for one issue for instance and the reasoning for specific PR changes isnt stated in the issue. Or alternatively, the issue might only state what is required, like delivering a feature, but you might in the PR need to do a lot of (related) work to get that feature completed.

File: PULL_REQUEST_TEMPLATE.md
Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md]: moved the linked issue from above to this section

File: PULL_REQUEST_TEMPLATE.md
Comment (ebezzi) [inline PULL_REQUEST_TEMPLATE.md:20]: Unless we make the QA process more formalized, I don't think this is particularly useful. Who will do the QA besides the engineer who submits the PR?

Comment (Bento007) [reply PULL_REQUEST_TEMPLATE.md:20]: I do sometimes depending on how big the changes are. I also only fill out this section when QA requires manual steps and it outside of simply running unit tests. The only time manual steps are usually needed is when it's db migration, or code that is outside of the application like admin scripts.

Comment (Bento007) [reply PULL_REQUEST_TEMPLATE.md:20]: I think this will most commonly be filled out with run the tests.

Comment (ebezzi) [reply PULL_REQUEST_TEMPLATE.md:20]: In that case, I feel like it's generally useful to coordinate with the oncall engineer so that the deployment goes smoothly. Since the oncall engineer isn't responsible for reviewing all PRs, filling the QA step here doesn't replace that.

Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md:20]: @ebezzi what do you mean by more formalized? 

This section is meant to indicate - how did you as the engineer test your code? How do you know its working? How would you test your code? And if its not something that requires testing, indicate why.

File: PULL_REQUEST_TEMPLATE.md
Comment (atolopko-czi) [inline PULL_REQUEST_TEMPLATE.md:2]: I don't think we use the functinoal vs readability reviewer designations consistently.  I'd be okay with eliminating this section entirely and just relying upon Github's reviewers list.  If a particular reviewer is needed for a particular _type_ of review, we can just note that in the Notes for Reviewer section, perhaps.

Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md:2]: sounds good to me i can remove this

File: PULL_REQUEST_TEMPLATE.md
Comment (atolopko-czi) [reply PULL_REQUEST_TEMPLATE.md:11]: I think we might use a better term than QA, since I think we're conflating this with PM/stakeholder QA testing.  Maybe just call this section "Testing".  I see this section as explaining to the reviewer how you made sure the change is ready to go.

  I would use this to call out things like "Unit tested only", "tested in rdev, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md:11]: @atolopko-czi @ebezzi @Bento007 thanks for the feedback here. I have updated the template, please let me know what you think

File: PULL_REQUEST_TEMPLATE.md
Comment (atolopko-czi) [inline PULL_REQUEST_TEMPLATE.md:6]: nit: could we put this section first? I find why more important than what.  And I want to be able to see the issue link right at the top as well.

**Review by atolopko-czi — APPROVED** [review]
LGTM, but requesting to put "Reason" section first.

File: PULL_REQUEST_TEMPLATE.md
Comment (dsadgat) [reply PULL_REQUEST_TEMPLATE.md:2]: done

**Review by Bento007 — APPROVED** [review]

**Pull request merged**
